packages/keycodes/src/index.js

```diff
@@ -42,7 +42,7 @@ import { isAppleOS } from './platform';
  *
  * @typedef {(character: string, isApple?: () => boolean) => T} WPKeyHandler
  */
-/** @typedef {(event: KeyboardEvent, character: string, isApple?: () => boolean) => boolean} WPEventKeyHandler */
+/** @typedef {(event: import('react').KeyboardEvent<HTMLElement>, character: string, isApple?: () => boolean) => boolean} WPEventKeyHandler */
```

Comment:
I _think_ this is what we want, given that it's meant to handle React synthetic events? Let me know if you disagree.

Reply:
Mhh, this is a tricky one.

For sure, the `onKeyDown` property returned by `useNavigateRegions` needs to take a `React.KeyboardEvent< HTMLDivElement >` as its argument.

Personally, I'm almost tempted to introduce a runtime change to the `navigateRegions` component and avoid making changes to the `@wordpress/keycodes` package, since from a quick look it seems that the package is not assuming react as a web dependency (only `react-native` is mentioned).

<details>
<summary>With this in mind, as an alternative approach we could use the `nativeEvent` property in `useNavigateRegions`</summary>

```diff
diff --git a/packages/components/src/higher-order/navigate-regions/index.tsx b/packages/components/src/higher-order/navigate-regions/index.tsx
index f2b9c31bfb..44afd39f4c 100644
--- a/packages/components/src/higher-order/navigate-regions/index.tsx
+++ b/packages/components/src/higher-order/navigate-regions/index.tsx
@@ -96,13 +96,19 @@ export function useNavigateRegions( shortcuts: Shortcuts = defaultShortcuts ) {
 		onKeyDown( event: React.KeyboardEvent< HTMLDivElement > ) {
 			if (
 				shortcuts.previous.some( ( { modifier, character } ) => {
-					return isKeyboardEvent[ modifier ]( event, character );
+					return isKeyboardEvent[ modifier ](
+						event.nativeEvent,
+						character
+					);
 				} )
 			) {
 				focusRegion( -1 );
 			} else if (
 				shortcuts.next.some( ( { modifier, character } ) => {
-					return isKeyboardEvent[ modifier ]( event, character );
+					return isKeyboardEvent[ modifier ](
+						event.nativeEvent,
+						character
+					);
 				} )
 			) {
 				focusRegion( 1 );
diff --git a/packages/keycodes/src/index.js b/packages/keycodes/src/index.js
index 15e96dca77..cd31fef5fa 100644
--- a/packages/keycodes/src/index.js
+++ b/packages/keycodes/src/index.js
@@ -42,7 +42,7 @@ import { isAppleOS } from './platform';
  *
  * @typedef {(character: string, isApple?: () => boolean) => T} WPKeyHandler
  */
-/** @typedef {(event: import('react').KeyboardEvent<HTMLElement>, character: string, isApple?: () => boolean) => boolean} WPEventKeyHandler */
+/** @typedef {(event: KeyboardEvent, character: string, isApple?: () => boolean) => boolean} WPEventKeyHandler */
 
 /** @typedef {( isApple: () => boolean ) => WPModifierPart[]} WPModifier */
 
@@ -346,7 +346,7 @@ export const shortcutAriaLabel = mapValues(
  * From a given KeyboardEvent, returns an array of active modifier constants for
  * the event.
  *
- * @param {import('react').KeyboardEvent<HTMLElement>} event Keyboard event.
+ * @param {KeyboardEvent} event Keyboard event.
  *
  * @return {Array<WPModifierPart>} Active modifier constants.
  */

```

</details>

Basically, trading off a small runtime change in exchange for not changing another package's APIs. What do you think?

Reply:
Thanks for digging deeper here! I looked into the codebase to see how `isKeyboardEvent()` is being used, and it is in fact handling both [native](https://github.com/WordPress/gutenberg/blob/6be1e861fbf7049935a3733a1d2061d360c079f9/packages/block-editor/src/components/rich-text/use-remove-browser-shortcuts.js#L24) and [synthetic](https://github.com/WordPress/gutenberg/blob/6be1e861fbf7049935a3733a1d2061d360c079f9/packages/block-library/src/navigation-submenu/edit.js#L293) events. The synthetic usages would all be errors if they were typechecked. So I think it would be best to correct the types so it matches the actual usage 👉 b94985a6

How does that sound?

Reply:
Sounds good to me!